File: Lab.Lenguaje de programacion/Actividad 2/Tarea2.c
#include <stdio.h>
#include <math.h>
#include <stdbool.h>
#include <string.h>

int main() {
    char buffer[1024];
    double numero, parteEntera, parteFraccionaria;
    bool entradaValida = false;

    // Solicitar al usuario un número hasta que la entrada sea válida
    while (!entradaValida) {
        printf("Ingrese un numero decimal: ");
        if (fgets(buffer, sizeof(buffer), stdin)) {
            // Verificar si la entrada contiene caracteres no deseados
            entradaValida = true;
            for (int i = 0; i < strlen(buffer); i++) {
                if ((buffer[i] < '0' || buffer[i] > '9') && buffer[i] != '.' && buffer[i] != '\n') {
                    entradaValida = false;
                    break;
                }
            }
            if (entradaValida) {
                // Convertir la entrada a un número decimal
                numero = atof(buffer);
                parteFraccionaria = modf(numero, &parteEntera);
                printf("Parte entera: %.0f\n", parteEntera);
                printf("Parte fraccionaria: %.2f\n", parteFraccionaria);
            } else {
                printf("Entrada invalida. Por favor ingrese solo numeros y un punto decimal.\n");
            }
        } else {
            // Error al leer la entrada
            printf("Error al leer la entrada. Intente de nuevo.\n");
            entradaValida = false;
        }
    }

    return 0;
}

File: Lab.Lenguaje de programacion/Actividad 2/Tarea2.py
def solicitar_numero_decimal():
    while True:
        entrada = input("Ingresa un número decimal: ")
        try:
            # Intenta convertir la entrada a un número flotante
            numero = float(entrada)
            # Verifica si la entrada es un número decimal (no entero)
            if '.' in entrada:
                return numero
            else:
                print("Error: Debes ingresar un número decimal, no un número entero.")
        except ValueError:
            # Se ejecuta si la conversión falla, lo que significa que no es un número
            print("Error: La entrada no es válida. Asegúrate de ingresar solo números decimales.")

# Solicita al usuario que ingrese un número decimal válido
numero_decimal = solicitar_numero_decimal()

# Separa el número en su parte entera y fraccionaria
parteEntera = int(numero_decimal)
parteFraccionaria = numero_decimal - parteEntera

# Imprime la parte entera y fraccionaria
print(f"Parte entera: {parteEntera}")
print(f"Parte fraccionaria: {parteFraccionaria:.2f}")
